### PR TITLE
Update Rust ToolBox

### PR DESCRIPTION
**Pull request by evolvedmicrobe**

### CODE REVIEW

**Comment by DavidBJaffe** [comment]
I think maybe this is failing because the gamma-delta changes are on rust-toolbox master, but not on enclone master.

**Comment by evolvedmicrobe** [comment]
jenkins retest this please

**Comment by evolvedmicrobe** [comment]
jenkins retest this please

**Comment by evolvedmicrobe** [comment]
jenkins retest this please

File: enclone_main/tmp.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Remove `vdj_asm_asm` as dependency of `cr_vdj`, make `vdj_asm_asm` binary actually do something.
+
+Inspired by https://github.com/10XDev/cellranger/pull/7629, and annoyed by some long compilations I was observing, I noticed that cr_vdj takes about 20 seconds to compile, and the vdj_asm_asm binary also takes about 20 seconds.  I wondered if I could remove the dependency of one on the other, and then also noticed that the vdj_asm_asm binary doesn't appear to be used anywhere.
+
+So this PR removes the dependency and makes the binary useful.
```
Comment (adam-azarchs) [inline enclone_main/tmp.txt:5]: Did you mean to commit this file?

Comment (evolvedmicrobe) [reply enclone_main/tmp.txt:5]: no, would need to remove this

**Comment by DavidBJaffe** [comment]
Nice! This makes a single-threaded test code use 4.8% less cycles.  This test only runs 
locally (on bespin1) and not in the CI because presumably it wouldn't work there.  The 
test would have failed locally because we don't allow any significant change, positive 
or negative.

The local test also includes a wall clock test on parallelized code, and the result of that did
not change.  However the single-threaded test uses the argument BUILT_IN, as does the
monster test that runs on Jenkins, so it is quite possible that the monster test (and similar
calculations) do indeed run faster.  In any case, good!

**Comment by evolvedmicrobe** [comment]
Ooh, super cool that you have a benchmark for instructions cycles setup!!!  That 5% number actually seems to match what it does for CI times to.  I compared the test times for this PR to the test times on the `master` branch and on Nima's recent PR.  The graphic below shows replicate runs of the test.  Looking at the minimum time observed it looks to have dropped that much.

image

The performance increase from this was less than I expected... life is always better if you don't test your hypotheses with data!  Just profiled again and it looks like I just moved the bottleneck to unpacking the other half of the seq being compared.  Gonna give it another try with that one removed as well.

**Comment by evolvedmicrobe** [comment]
jenkins retest this please

@DavidBJaffe `enclonetest4` looks cool, but I couldn't get it to work.  Gonna abort on further improvements and move on.

**Comment by evolvedmicrobe** [comment]
Merged the change to rust toolbox, I'm gonna close this and let the change come in whenever `vdj_ann` is updated

**Pull request closed without merging**